docs/user/developing_jobs.md

```diff
@@ -6,43 +6,96 @@ Familiarity with [DiffSync](https://diffsync.readthedocs.io/en/latest/) and with
+1. Define the base models that define the data contract
+2. Implement the Nautobot adapter automatically through `nautobot_ssot.contrib.BaseAdapter`
+3. If Nautobot is the source of the job implement `create`, `update`, `delete` methods for the models in the remote system
+4. Implement the remote adapters `load` method, responsible for loading the data from the remote system into the adapter 
+5. Putting it together into a job
 
-   - Define your models based on the data you want to sync. For example, if you are syncing specific attributes (e.g. tags) of a device, but not the devices themselves, your parent model
-     should be the tags. This approach prevents unnecessary `create`, `update`, and `delete` calls for models that are not sync'd.
-   - For each model class, implement the `create`, `update`, and `delete` DiffSyncModel APIs for writing data to the Data Target system.
+### Defining the Models
 
-2. Define a `DiffSync` adapter class for loading initial data from Nautobot and constructing instances of each `DiffSyncModel` class to represent that data.
-3. Define a `DiffSync` adapter class for loading initial data from the Data Source or Data Target system and constructing instances of the `DiffSyncModel` classes to represent that data.
+The models use diffsync, which in turn uses [pydantic](https://docs.pydantic.dev/latest/) for its data modelling. Nautobot SSoT comes with a set of classes out of the box that implement a lot of functionality for you automatically using the classes from `nautobot_ssot.contrib`, provided you adhere to a set of rules.
 
-4. Develop a Job class, derived from either the `DataSource` or `DataTarget` classes provided by this plugin, and implement the adapters to populate the `self.source_adapter` and `self.target_adapter` that are used by the built-in implementation of `sync_data`. This `sync_data` method is an opinionated way of running the process including some performance data, more in [next section](#analyze-job-performance), but you could overwrite it completely or any of the key hooks that it calls:
+The first rule is to define your models tightly after the Nautobot models themselves. Example:
 
-   - `self.load_source_adapter`: This is mandatory to be implemented. As an example:
+```python
+from nautobot.tenancy.models import Tenant
+from nautobot_ssot.contrib import NautobotModel
+
+class TenantModel(NautobotModel):
+    """An example model of a tenant."""
+    _model = Tenant
+    _modelname = "tenant"
+    _identifiers = ("name",)
+    _attributes = ("description",)
+
+    name: str
+    description: str
+```
+
+As you can see when looking at the source code of the `nautobot.tenancy.models.Tenant` model, the fields on the model (`name` and `description`) match the names of the fields on that model one-to-one. This enables the base class `NautobotModel` to dynamically load, create, update and delete your tenants without you needing to implement this functionality yourself.
```

Comment:
```suggestion
As you can see when looking at the source code of the `nautobot.tenancy.models.Tenant` model, the fields on the model (`name` and `description`) match the names of the fields on that model exactly. This enables the base class `NautobotModel` to dynamically load, create, update and delete your tenants without you needing to implement this functionality yourself.
```

Reply:
one-to-one is a term used in django orm and databases, my mind goes to that definition.

Reply:
why if I want to reimplement the functionality to add some customization? maybe a note to show that this is possible?